BMM.Core/Implementations/DeepLinking/DeepLinkHandler.cs

```diff
@@ -71,12 +75,13 @@ IUserAuthChecker authChecker
                 new RegexDeepLink("^/music$", NavigateTo<ExploreRecentMusicViewModel>),
                 new RegexDeepLink("^/contributors$", NavigateTo<ExploreContributorsViewModel>),
                 new RegexDeepLink("^/featured$", NavigateTo<CuratedPlaylistsViewModel>),
-                new RegexDeepLinkWithParameters("^/playlist/curated/(?<id>[0-9]+)(/(?<name>.*))?$", OpenCuratedPlaylist),
-                new RegexDeepLinkWithParameters("^/playlist/private/(?<id>[0-9]+)(/(?<name>.*))?$", OpenTrackCollection),
-                new RegexDeepLinkWithParameters("^/playlist/contributor/(?<id>[0-9]+)(/(?<name>.*))?$", OpenContributor),
-                new RegexDeepLinkWithParameters("^/playlist/podcast/(?<id>[0-9]+)(/(?<name>.*))?$", OpenPodcast),
-                new RegexDeepLinkWithParameters("^/album/(?<id>[0-9]+)$", OpenAlbum),
-                new TrackLinkParser("^/track/(?<id>[0-9]+)(/(?<language>.*))?$", (id, name, startTimeInMs) => PlayTrackById(id, startTimeInMs)),
+                new RegexDeepLink<StandardDeepLinkParameters>("^/playlist/curated/(?<id>[0-9]+)(/(?<name>.*))?$", OpenCuratedPlaylist),
```

Comment:
I think the names "StandardDeepLinkParameters" and "TrackCollectionDeepLinkParameters" can be improved

Reply:
I aggree with "SharingSecretParameter" but not with "IdAndNameParameter". What if in the future we add another value that is used there? Will we rename it to "IdAndNameAnd(...)Parameter"? ;)
I named it Standard, because it is the most used set of parameters in deep links iin our case.

Reply:
But why would we add another parameter? It seems very unlikely that we add a parameter to all playlists at once. If we need another parameter for podcast, then we would need another IDeepLinkParameter class and then we can pick the name that makes sense. But the existing class IdAndNameParameter would stay the same.